Content.Server/StationEvents/EventManagerSystem.cs

```diff
@@ -60,6 +63,71 @@ public string RunRandomEvent()
         return FindEvent(availableEvents);
     }
 
+    /// <summary>
+    /// Randomly runs an event from provided EntitySpawnCollection.
+    /// </summary>
+    public string? RunLimitedEvent(List<EntitySpawnEntry>? limitedEventsList)
+    {
+        if (limitedEventsList != null)
+        {
+            var availableEvents = AvailableEvents(); // handles the player counts and individual event restrictions
+
+            if (availableEvents.Count == 0)
+            {
+                Log.Warning("No events were available to run!");
+                return null;
+            }
```

Comment:
Having nullable APIs like this is bad, just add overloads.

Reply:
ah smort. I always forget about overloads.